app/controllers/unattended_controller.rb

```diff
@@ -51,6 +54,25 @@ def hostgroup_template
   # i.e. /unattended/provision will render the provisioning template for the requesting host
   def host_template
     return head(:not_found) unless params[:kind].present?
+
+    if ipxe_request? && !@host && params[:bootstrap]
```

Comment:
I wouldn't nest these. Much harder to read imo.

Good catch on the return. I don't believe the what the method returns since it is just rendering the template. I guess more syntactical sugar.

Reply:
To me the nesting makes it's clearer there is one workflow. If you have single `return render_` lines I think it can be easier to understand the ipxe workflow.

Reply:
I disagree. Having nested if statements makes it hard to read the code, and it takes longer to understand what is happening.  

Having early returns makes the when and why much much clearer to the reader. Rubocop style guid also advises against nested if statements. https://github.com/rubocop-hq/ruby-style-guide#no-nested-conditionals

Reply:
I know this is not helpful: I don't like both. Can we extract this further? E.g. a class that finds the appropriate template based on conditions and then just renders that? What do you think?

Reply:
To make it clearer:

We have a lot of methods like
```ruby
template = ProvisioningTemplate.where(...)
if template
   sefe_render(template)
else
  render_error
end
```

I think we should clean this up and then this will go away.

A service that finds the appropriate template for a request.

Reply:
I think that is getting too extreme. Rendering should be in the controller and not in an external class. 

Due to the complexity of the unattended controller there is no way of avoiding all fo the early returns. The controller is responsible for too many things. Sure its all about unattended installations but maybe we can think about splitting this one up into smaller, more focused controllers.... OR we can just accept complexity and keep the early returns and try and keep the code as readable as possible.

However I would stay well clear of nested if statements or extracting out the `render` calls and logic to an external class. 

I agree however that this can be dried up:
```ruby
if template
   sefe_render(template)
else
  render_error
end
```

Reply:
> Rendering should be in the controller and not in an external class.

I fully agree, but we could create a service that finds the correct template for a request. The controller can then do the rendering. WDYT?

Reply:
Even if it's a helper function within the controller it does make sense to me.

Reply:
This is a step in the right direction. The only hard part is managing the errors that are rendered.

I had a go at cleaning it up but the methods `sefe_render(template)` and `render_custom_error(status, error_message, params)` made it hard. These two should be cleaned up and merged into one but this will require deeper thought and more time.

This is all good discussion but theres one problem. Any changes now will cause us(me) pain with merge conflicts. So lets stay focused on the introduced changes. I would say we focus on  #6009, get it merged. Then the changes here, and then I can look at cleaning up the logic for finding the templates by extracting it out of the controller. sound ok @timogoebel @ekohl  ?